SendLog.py

```diff
@@ -31,38 +40,48 @@
 ENDING = 1
 DATA = 2
 NOLINE = -1
+BLOCK_IDENTIFICATION =- 2
+
 # Scilab dir, can't run absolute paths
-#SCI = "../scilab/"
-#SCI="../scilab-master/scilab/"
-SCI="../../scilab-master_5.5.2/"
+SCI = "../scilab_for_xcos/"
+
 # List to store figure IDs
 figure_list = []
 # List to store filenames of files
 xcos_file_list = []
+#list to identify whether to save to workspace ,or load from workspce or neither
+workspace_list =[]
+#dictionary to find variable to load or save from workspace
+workspace_dict ={}
+
 
 class line_and_state:
-	# Class to store the line and its state
-	line = None
-	state = NOLINE
-	def __init__(self, line, state):
-		self.line = line
-		self.state = state
-	def set(self, line_state):
-		self.line = line_state[0]
-		self.state = line_state[1]
-		return False
-	def get_line(self):
-		return self.line
-	def get_state(self):
-		return self.state
-		
+    # Class to store the line and its state
+    line = None
+    state = NOLINE
+    def __init__(self, line, state):
+        self.line = line
+        self.state = state
+    def set(self, line_state):
+        self.line = line_state[0]
+        self.state = line_state[1]
+        return False
+    def get_line(self):
+        return self.line
+    def get_state(self):
+        return self.state
+        
 def parse_line(line):
-	# Function to parse the line
+   	# Function to parse the line
 	# Returns tuple of figure ID and state
 	# state = INITIALIZATION if new figure is created
 	#         ENDING if current fig end
 	#         DATA otherwise
 	line_words = line.split(' ')
+        #The below condition determines the block ID 
+        if line_words[2] == "Block":
+                block_id=int(line_words[4])
```

Comment:
You have used 4-space length indents above. The code here is now using 8-space length indents. Please fix and maintain a consistent indentation length all across the code.